scripts/build_demo_app.py

```diff
@@ -0,0 +1,242 @@
+
+    files = ["re_viewer_bg.wasm", "re_viewer_debug_bg.wasm", "re_viewer.js", "re_viewer_debug.js"]
+    for file in files:
+        shutil.copyfile(
+            os.path.join("web_viewer", file),
+            os.path.join(BASE_PATH, file),
+        )
+
+
+def collect_examples() -> List[Example]:
+    examples = []
+    for name in EXAMPLES.keys():
+        example = Example(
+            name,
+            title=EXAMPLES[name]["title"],
+            description=EXAMPLES[name]["description"],
+        )
+        if example.supports_save():
+            examples.append(example)
+    return examples
+
+
+def save_examples_rrd(examples: List[Example]) -> None:
+    logging.info("\nSaving examples as .rrd")
+
+    for example in examples:
+        example.save()
+
+
+def render_examples(examples: List[Example]) -> None:
+    logging.info("\nRendering examples")
+
+    template_path = os.path.join(SCRIPT_PATH, "demo_assets/templates/example.html")
+    with open(template_path) as f:
+        template = Template(f.read())
+
+    for example in examples:
+        index_path = f"{BASE_PATH}/examples/{example.name}/index.html"
+        logging.info(f"{example.name} -> {index_path}")
+        with open(index_path, "w") as f:
+            f.write(template.render(example=example, examples=examples))
+
+
+def serve_files(port: int) -> None:
+    def serve() -> None:
+        logging.info(f"\nServing examples at http://localhost:{port}/")
+        server = http.server.HTTPServer(
+            server_address=("127.0.0.1", port),
+            RequestHandlerClass=partial(
+                http.server.SimpleHTTPRequestHandler,
+                directory=BASE_PATH,
+            ),
+        )
+        server.serve_forever()
+
+    threading.Thread(target=serve, daemon=True).start()
+
+
+def main() -> None:
+    logging.getLogger().addHandler(logging.StreamHandler())
+    logging.getLogger().setLevel("INFO")
+
+    parser = argparse.ArgumentParser(description="Build and/or serve `demo.rerun.io`")
+    parser.add_argument(
+        "--serve",
+        type=int,
+        default=8080,
+        action="store",
+        nargs="?",
+        help="Serve the app on this port after building [default: 8080]",
+    )
+
+    args, unknown = parser.parse_known_args()
+    for arg in unknown:
+        logging.warning(f"unknown arg: {arg}")
+
+    copy_static_assets()
+    build_and_copy_wasm()
+    examples = collect_examples()
+    save_examples_rrd(examples)
+    render_examples(examples)
+
+    if "serve" in args:
+        port = args.serve or 8080
+
+        serve_files(port)
+
+        while True:
+            try:
+                logging.info("Press enter to reload static files")
+                input()
+                copy_static_assets()
+                render_examples(examples)
+            except KeyboardInterrupt:
+                break
+
+
+BASE_PATH = "web_demo"
+SCRIPT_PATH = os.path.dirname(os.path.relpath(__file__))
+EXAMPLES = {
```

Comment:
I spent some time overengineering this part, but then changed it to just store the example list here. We're cherry-picking the examples anyway, so I think it makes sense to also write the title and description manually.

Not sure if it should be stored directly in the script. I don't think there's much of a downside, though.

Reply:
Talked about this in the standup, this is fine, at least for now.